### DISCUSSION TITLE
IKEA	E1757 sends data request every second to TS011F_din_smart_relay (probably an issue)

### DISCUSSION THREAD

**Discussion by PietPetrus** [Koenkk/zigbee2mqtt#12969] [Devices]
My Ikea cover drops out randomly, meaning it does not respond to the remote nor to commands directly from the coordinator.
Did the usual battery/network/repairing/sanity check and finally decided to sniff the packets.

It looks like the cover (0xe352) is sending a data request every second to the TS011F_din_smart_relay (0x29d7) with destination PAN 0x1a63.

In the meantime the cover is completely unresponsive to commands.

I cannot see packets being sent from 0x29d7 with data from 0xe352, but I do not know if that's to be expected.

1. Is it expected behaviour that the E1757 sends a data request every second containing the following bytes (example below)?
2. Is the problem that the TS011F_din_smart_relay is not relaying any messages or responding?
3. Is there a way to enforce a different route to see if that solves the problem?
4. How can I solve this 😄 ?

```
63 88 b3 63 1a d7 29 52 e3 04                     c..c..)R..

IEEE 802.15.4 Command, Dst: 0x29d7, Src: 0xe352 
Frame Control Field: 0x8863, Frame Type: Command, Acknowledge Request, PAN ID Compression, Destination Addressing Mode: Short/16-bit, Frame Version: IEEE Std 802.15.4-2003, Source Addressing Mode: Short/16-bit
        .... .... .... .011 = Frame Type: Command (0x3)
        .... .... .... 0... = Security Enabled: False
        .... .... ...0 .... = Frame Pending: False
        .... .... ..1. .... = Acknowledge Request: True
        .... .... .1.. .... = PAN ID Compression: True
        .... .... 0... .... = Reserved: False
        .... ...0 .... .... = Sequence Number Suppression: False
        .... ..0. .... .... = Information Elements Present: False
        .... 10.. .... .... = Destination Addressing Mode: Short/16-bit (0x2)
        ..00 .... .... .... = Frame Version: IEEE Std 802.15.4-2003 (0)
        10.. .... .... .... = Source Addressing Mode: Short/16-bit (0x2)
    Sequence Number: 96
    Destination PAN: 0x1a63
    Destination: 0x29d7
    Source: 0xe352
    Command Identifier: Data Request (0x04)

```

**Comment by Koenkk** [comment]
That is expected; the way sending messages to end devices works is as follows:
- Given that we have coordinator, router and an end device in the network
- Coordinator sends a command to the end device
- Router see the message passing by and holds it (only if the end device is a direct child of the router)
- End device wakes up and does a "data request" to the router (asking like; do you have any messages for me?)
- if yes, routers sends it

It may be that the TS011F router is not correctly relaying messages to the cover.

**Reply by PietPetrus** [reply]
So I should start a sniff, re-pair the cover, wait until the cover isn't reached by the coordinator?

Could it be that the Tuya device announces itself as a router but never declares or acts upon the new route?

Disregarding messages from the coordinator and/or not relaying messages from the end device to the coordinator?

Thanks for the quick reply though! 😊

**Reply by Koenkk** [reply]
> So I should start a sniff, re-pair the cover, wait until the cover isn't reached by the coordinator?

I think that is the easiest way to see what goes wrong (because you can compare the working and non-working situation)
